pkg/daemon/ceph/osd/volume.go

```diff
@@ -1158,10 +1158,17 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 
 		if !skipDeviceClass {
 			diskInfo, err := clusterd.PopulateDeviceInfo(blockPath, context.Executor)
```

Comment:
We can move this line after checking `if osd.DeviceClass == "" {` line.

Reply:
I have updated the code please check